File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.java
class Solution {
    public int characterReplacement(String s, int k) {
        int longest = 0;
        int[] count = new int[26];
        int l = 0;
        int r = 0;

        while (r < s.length()) {
            char c = s.charAt(r);
            count[c - 'A']++;
            int currentMaxCount = 0;
            for (int num : count) {
                currentMaxCount = Math.max(currentMaxCount, num);
            }

            int winLength = r - l + 1;
            if (winLength - currentMaxCount > k) {
                count[s.charAt(l) - 'A']--;
                l++;
            } else {
                longest = Math.max(longest, winLength);
            }

            r++;
        }

        return longest;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # 10 9 2 1 5
        # 1
        # 1 2
        maxProfit = 0
        l, r = 0, 1

        while r < len(prices):

            maxProfit = max(maxProfit, prices[r] - prices[l])
            if prices[l] >= prices[r]:
                l = r

            r += 1
        
        return maxProfit